code/game/objects/item.dm

```diff
@@ -475,19 +475,12 @@
 	if(!inv_slot)
 		return FALSE
 
-	var/already_equipped = inv_slot.get_equipped_item()
-	if(!ignore_equipped || already_equipped != src)
-		if(already_equipped)
-			if(!force)
-				return FALSE
-			inv_slot.clear_slot()
-			qdel(already_equipped)
-
-		if(!inv_slot.is_accessible(M, src, disable_warning))
+	if(!force)
+		if(!ignore_equipped && !inv_slot.is_accessible(M, src, disable_warning))
```

Comment:
As mentioned on Discord, my implementation for ``ignore_equipped`` was only intended to ignore blocking from the src item if it was equipped, not blocking from all items equipped. This was why the if statement had ``||`` rather than ``&&``. It's probably okay to change this, but the if statements in the overrides on gloves and magboots should be changed as well.